### PR TITLE
add `reset_threads!()`

### PR DESCRIPTION
**Pull request by ranocha**
I added a function `reset_threads!()` to reset the threads after throwing an exception as discussed in #30.

### CODE REVIEW

File: src/Polyester.jl [hunk 1]
```diff
@@ -25,6 +26,21 @@ export batch, @batch, disable_polyester_threads
...
+# see https://github.com/JuliaSIMD/Polyester.jl/issues/30
+"""
+    Polyester.reset_threads!()
+
+Resets the threads used by [Polyester.jl](https://github.com/JuliaSIMD/Polyester.jl).
+"""
+function reset_threads!()
+  PolyesterWeave.reset_workers!()
+  for i in 1:(PolyesterWeave.dynamic_thread_count() - 1)
+      ThreadingUtilities.initialize_task(i)
```
Comment (chriselrod) [inline src/Polyester.jl:39]: I would only reinitialize if not running.

Comment (ranocha) [reply src/Polyester.jl]: As implemented in 514b7ac?

**Comment by chriselrod** [comment]
Thanks!

**Comment by ranocha** [comment]
Thanks you for all your effort you put into a nice performance ecosystem!

**Pull request merged**